beacon_node/eth1_chain/src/web3_fetcher.rs

```diff
@@ -0,0 +1,245 @@
+use bls::{PublicKeyBytes, SignatureBytes};
+use ethabi::{decode, ParamType, Token};
+use parking_lot::RwLock;
+use std::collections::BTreeMap;
+use std::marker::Send;
+use std::sync::Arc;
+use types::DepositData;
+use web3::contract::{Contract, Options};
+use web3::futures::{Future, Stream};
+use web3::transports::WebSocket;
+use web3::types::FilterBuilder;
+use web3::types::*;
+use web3::Web3;
+
+use crate::types::{ContractConfig, Eth1DataFetcher};
+
+/// Wrapper around web3 api.
+#[derive(Clone, Debug)]
+pub struct Web3DataFetcher {
+    event_loop: Arc<web3::transports::EventLoopHandle>,
+    /// Websocket transport object. Needed for logs subscription.
+    web3: Arc<web3::api::Web3<web3::transports::ws::WebSocket>>,
+    /// Deposit contract config.
+    contract: ContractConfig,
+}
+
+impl Web3DataFetcher {
+    /// Create a new Web3 object.
+    pub fn new(endpoint: &str, deposit_contract: ContractConfig) -> Web3DataFetcher {
+        let (event_loop, transport) = WebSocket::new(endpoint).unwrap();
```

Comment:
Can we handle the case where the `Eth1` server goes offline? E.g., reconnects?

Reply:
Not currently. Relevant PR in the `web3` crate https://github.com/tomusdrw/rust-web3/issues/151